quinn-proto/src/connection/stats.rs

```diff
@@ -0,0 +1,80 @@
+/// Statistics about packets transmitted on a connection
+#[derive(Default)]
+pub struct PacketTransmissionStats {
+    /// The amount of packets transmitted on a connection
+    pub packets: u64,
+    /// The total amount of bytes transmitted on a connection
+    pub bytes: u64,
+}
+
+impl PacketTransmissionStats {
+    /// Calculates the average packet size
+    pub fn avg_packet_size(&self) -> u64 {
+        if self.packets != 0 {
+            self.bytes / self.packets
+        } else {
+            0
+        }
+    }
+}
+
+impl std::fmt::Debug for PacketTransmissionStats {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("PacketTransmissionStats")
+            .field("number of packets", &self.packets)
+            .field("number of bytes", &self.bytes)
+            .field("avg packet size", &self.avg_packet_size())
+            .finish()
```

Comment:
I feel like a more conventional, perhaps derived, `Debug` impl would be fine here. Not sure `Debug` impls that produce invalid syntax is a good habit.

Reply:
How is it invalid syntax?

Reply:
In the sense that `number of bytes` is not an identifier.

Reply:
I don’t think Debug really cares about it. It produces arbitrary syntax. And even if we go towards „it’s a regular Json-like syntax“, the whitespace still seems ok. 
Anyway, I can change it to have underscores if people care about it

Reply:
It doesn't really care; it's mostly a stylistic question, in that it's inconsistent with most other `Debug` impls. I also don't think "number of packets: 42" is actually any clearer than "packets: 42".